Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp

```diff
@@ -0,0 +1,229 @@
+
+	GaussianSplatComponent->SetScale(csp::common::Vector3(2.0f, 4.0f, 6.0f));
+	EXPECT_EQ(GaussianSplatComponent->GetScale(), csp::common::Vector3(2.0f, 4.0f, 6.0f));
+
+	GaussianSplatComponent->SetIsVisible(false);
+	EXPECT_EQ(GaussianSplatComponent->GetIsVisible(), false);
+
+	GaussianSplatComponent->SetIsARVisible(false);
+	EXPECT_EQ(GaussianSplatComponent->GetIsARVisible(), false);
+
+	GaussianSplatComponent->SetIsShadowCaster(false);
+	EXPECT_EQ(GaussianSplatComponent->GetIsShadowCaster(), false);
+
+	GaussianSplatComponent->SetTint(csp::common::Vector3(1.0f, 0.4f, 0.0f));
+	EXPECT_EQ(GaussianSplatComponent->GetTint(), csp::common::Vector3(1.0f, 0.4f, 0.0f));
+
+	SpaceSystem->ExitSpace([](const csp::systems::NullResult& Result){});
+
+	// Delete space
+	DeleteSpace(SpaceSystem, Space.Id);
+
+	// Log out
+	LogOut(UserSystem);
+}
+#endif
+
+#if RUN_ALL_UNIT_TESTS || RUN_GAUSSIAN_SPLAT_TESTS || RUN_GAUSSIAN_SPLAT_SCRIPT_INTERFACE_TEST
+CSP_PUBLIC_TEST(CSPEngine, GaussianSplatTests, GaussianSplatScriptInterfaceTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* Connection				 = SystemsManager.GetMultiplayerConnection();
+	auto* EntitySystem				 = SystemsManager.GetSpaceEntitySystem();
+
+	const char* TestSpaceName		 = "OLY-UNITTEST-SPACE-REWIND";
+	const char* TestSpaceDescription = "OLY-UNITTEST-SPACEDESC-REWIND";
+
+	char UniqueSpaceName[256];
+	SPRINTF(UniqueSpaceName, "%s-%s", TestSpaceName, GetUniqueString().c_str());
+
+	// Log in
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	// Create space
+	csp::systems::Space Space;
+	CreateSpace(SpaceSystem, UniqueSpaceName, TestSpaceDescription, csp::systems::SpaceAttributes::Private, nullptr, nullptr, nullptr, Space);
+
+	auto [EnterResult] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+
+	EXPECT_EQ(EnterResult.GetResultCode(), csp::systems::EResultCode::Success);
+
+	EntitySystem->SetEntityCreatedCallback(
+		[](csp::multiplayer::SpaceEntity* Entity)
+		{
+		});
+
+	// Create object to represent the image
+	csp::common::String ObjectName = "Object 1";
+	SpaceTransform ObjectTransform = {csp::common::Vector3::Zero(), csp::common::Vector4::Zero(), csp::common::Vector3::One()};
+	auto [CreatedObject]		   = AWAIT(EntitySystem, CreateObject, ObjectName, ObjectTransform);
+
+	auto* GaussianSplatComponent = (GaussianSplatSpaceComponent*) CreatedObject->AddComponent(ComponentType::GaussianSplat);
+	auto* ScriptComponent = (ScriptSpaceComponent*) CreatedObject->AddComponent(ComponentType::ScriptData);
+
+	CreatedObject->QueueUpdate();
+	EntitySystem->ProcessPendingEntityOperations();
+
+	EXPECT_EQ(GaussianSplatComponent->GetIsVisible(), true);
+
+	// Setup script
+	const std::string ScriptSource = R"xx(
+	
+		var splat = ThisEntity.getGaussianSplatComponents()[0];
+		splat.tint = [0.0, 0.1, 0.2];
+    )xx";
```

Comment:
1 tab missing?

Reply:
No, more CSP clang formatter work on display. It reads fine in-IDE. As designed.